### PR TITLE
[CN-536] Refactor datastructure CRs

### PR DESCRIPTION
**Pull request by cagric0**
Refactor
- Multimap and Topic CRs
- Hazelcast configMap creation
- Dependent CR deletion

Add DataStructure controller and status

### CODE REVIEW

File: test/e2e/helpers_test.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 	"context"
 	"encoding/xml"
 	"fmt"
+	"github.com/hazelcast/hazelcast-platform-operator/controllers/hazelcast"
```
Comment (SeriyBg) [inline test/e2e/helpers_test.go:9]: Could we order the imports idiomatically like this:
```
import (
    stdlib

    current_project

    company

    all others 
)
```

**Review by SeriyBg — APPROVED** [review]
Just a small nit, other than that LGTM

File: test/e2e/helpers_test.go (hunk 1 above)
Comment (cagric0) [reply test/e2e/helpers_test.go:9]: As far as I can see from other files, the order is a bit different;

```
import (
    stdlib

    company
    k8s

    current_project
)
```

Do you want me to change them on all the files? I think this should be done in a seperate PR. For now I'll just order the imports of [test/e2e/helpers_test.go](https://github.com/hazelcast/hazelcast-platform-operator/pull/393/files/c883be863d2fb5ad75b329f9ec44805763319c16#diff-0a46baa7716e5cf9a396958c52643c1515750e55f917c4e20c3cf51991514707) according to other files' structure. Is that okay for you?

**Review by mtyazici — APPROVED** [review]
LGTM

**Pull request merged**